src/frontend/src/views/Data/components/TreeTableNameCell/index.tsx

```diff
@@ -18,12 +31,55 @@ interface Props {
   item: PhyloRun;
 }
 
+const getDateRangeString = (item: PhyloRun): string => {
+  // NOTE: The start date default is covid-specific. We will need to update for
+  // other pathogens
+  const startDate = item.templateArgs?.filterStartDate || "2019-12-23";
+  const endDate =
+    item.templateArgs?.filterEndDate || item.startedDate.slice(0, 10);
+
+  return `${startDate} to ${endDate}`;
+};
+
 const TreeTableNameCell = ({ value, item }: Props): JSX.Element => {
   const [open, setOpen] = useState(false);
   const { phyloTree, status, user } = item;
   const treeId = phyloTree?.id;
   const userName = user?.name;
   const isDisabled = status !== TREE_STATUS.Completed || !treeId;
+  const { data: namedLocationsData } = useNamedLocations();
+
+  const { data: groupInfo } = useGroupInfo();
+  const { namedLocationsById } = useMemo(() => {
+    if (!namedLocationsData) return {};
+    return {
+      namedLocationsById: reduceObjectArrayToLookupDict(
```

Comment:
Nit: Doing this reduce operation for every single tree name cell seems kind of expensive, especially if we have a lot of tree rows in the table. We just need the same lookup object for all of them, right?

This might definitely be in the category of premature optimization, so grain of salt and all that, but it would be nice if we could create this at a higher level one time and just pass down the same lookup object to all of the name cells. Counter-point: getting that to work with our current table architecture would probably be a real ugly pain. The parent to this -- if I recall correctly -- is very dynamic, so inserting it in a sane way would be tough.

Alternatively, you could create a new react-query for locations, ala the existing `useNamedLocations`. But instead if would have a more complicated `select` function to generate the lookup. I'm _pretty sure_ that would only cause the lookup generation to run once, even if all the cells are calling the new `useLocationsLookup`, but... double check me on that and verify the `select` really does only run once.

All that said, if you don't think it's worth the hassle / don't have the time, don't worry. But if you feel like you can spare the time, it seems like a reasonable nice to have: the locations response is reasonably big, like ~0.5MB and ~35k location objects in there (those numbers are from me checking Staging).

Reply:
Good call.  Adding a `useNamedLocationsById` select function was easy enough